jep-0000/README.adoc

```diff
@@ -0,0 +1,153 @@
+Due to its design, it is not always possible to apply corresponding changes to ruby-runtime itself.
+Instead, every dependent plugin may need to be adapted individually, if a change to core results in problems to ruby-runtime.
+
+There currently are no plugins based on ruby-runtime that are both actively maintained (a release in the last two years, i.e. since June 2016) and popular (>2500 reported installations).
+In a Jenkins developers mailing list thread proposing the deprecation of `ruby-runtime` in May 2018, nobody volunteered to maintain `ruby-runtime` and address its problems.
+
+Therefore this JEP proposed the deprecation of ruby-runtime, and to suspend its distribution via Jenkins project update sites.
+
+
+== Reasoning
+
+We have a limited set of tools available to deal with problems like this.
+
+* Security warnings don't really apply here, as UI labels specifically mention 'security'.
+* Wiki pages and other plugin documentation is easy to overlook, especially in automated deployments.
+* There are no ways for users to provide feedback for plugins like there is for core (via changelog 'weather').
+* There are no ways to mark plugins as incompatible and, for example, warn users on upgrade, other than to suspend distribution. Additionally, we'd need to test every core change for ruby-runtime impact to know of the problem in advance.
+
+So the viable options are the following:
+
+* We could continue to distribute ruby-runtime while reverting the changes in Jenkins core that make it work. This will just result in a bad user experience, as ruby-runtime based plugins remain available while not working with new Jenkins releases.
+* We could continue to distribute ruby-runtime and keep the already implemented changes to core around, hoping no further problems occur. If they do, we can still implement this proposed deprecation of ruby-runtime. In this case, there would be no advance warning of current ruby-runtime users, and the number of users may increase in the mean time, making it more difficult to justify such a change.
+* We could continue to distribute ruby-runtime, keep the already implemented changes to core around, and fix any future problems. This option comes with potentially significant work with very little benefit, as ruby-runtime based plugins are neither very popular, nor actively maintained.
+
+
+== Backwards Compatibility
+
+Existing users can continue to use ruby-runtime based plugins.
+ruby-runtime and plugins depending on it can still be downloaded from Artifactory to support legacy environments.
+
+If previous core compatibility fixes are reverted, or future core changes break ruby-runtime, users of those plugins will be impacted.
+
+
+== Security
+
+There are no security risks related to this proposal.
+
+
+== Infrastructure Requirements
+
+This JEP will be implemented by using a well established feature of the update center generator.
+
+There are no new infrastructure requirements related to this proposal.
```

Comment:
Unless we implement the new-update site proposal in the thread

Reply:
it would be also great to still list these plugins in plugins.jenkins.io (with a "deprecated" warning and link to the download guidelines)

Reply:
It absolutely would be, but it's not possible with what we have.